### DISCUSSION TITLE
How do I search my table?

### DISCUSSION THREAD

**Discussion by Slddev** [supabase/supabase#22348] [Questions]
I want to search my table with it returning all records that remotely looks like the query, for example:

If I have a table `countries` with `[Pakista, Papua New Guinea, Peru]`, if my query is `p` it should return all of those records, or if I query `pa` it should return `[Pakista, Papua New Guinea]` and etc. 

I tried using full text search, the `.ilike` filter but they only return if it matches the whole string. Any help is very useful!

**Comment by GaryAustin1** [comment]
The ilike filter has partial searching with a wildcard % at start or end of string.  https://supabase.com/docs/reference/javascript/ilike
Full text search use lexumes which don't work well with partial searches.  If ilike partial matching is not "fuzzy" enough there are also trigram searches.  https://www.postgresql.org/docs/current/pgtrgm.html
